numpy/core/src/multiarray/mapping.c

```diff
@@ -2528,13 +2524,17 @@ PyArray_MapIterCheckIndices(PyArrayMapIterObject *mit)
         op_iter = NpyIter_New(op,
                         NPY_ITER_BUFFERED | NPY_ITER_NBO | NPY_ITER_ALIGNED |
                         NPY_ITER_EXTERNAL_LOOP | NPY_ITER_GROWINNER |
-                        NPY_ITER_READONLY,
+                        NPY_ITER_READONLY | NPY_ITER_ZEROSIZE_OK,
                         NPY_KEEPORDER, NPY_SAME_KIND_CASTING, intp_type);
 
         if (op_iter == NULL) {
             Py_DECREF(intp_type);
             return -1;
         }
+        if (NpyIter_GetIterSize(op_iter) == 0) {
+            NpyIter_Deallocate(op_iter);
+            continue;
+        }
```

Comment:
I am curious to know why above lines are not tested. Is it because the array not being trivially iterable is not an important use case ? my understanding is trivial iterable (from looking at the macros) means both F_CONTIGUOUS and C_CONTIGUOUS flags are false and ndim > 1. I was looking for an example for this, but wasnt able to find it. would you be able to point me to it.

Reply:
Not sure I follow, did you try coverage of thes elines and they are never hit? Now that you are saying this, it is a bit surprising that these are necessary, and most likely they are effectively never necessary, because we should not have `NPY_RELAXED_STRIDES_SETTING=0` anymore (except in the tests currently). So that effectively, these lines should never be hit.
Trivially iterable should be contiguous (F or C) *or* 1-D. And empty arrays are always contiguous with `NPY_RELAXED_STRIDES_SETTING=1`...

Maybe we should actually remove this here with a comment and just get rid of the environment variable for good...

Reply:
> Not sure I follow, did you try coverage of thes elines and they are never hit
yes i tried it with gdb. these lines never hit for the test cases.

> Trivially iterable should be contiguous (F or C) or 1-D.
Maybe this is a stupid question, Can there be an array which is neither C nor F contiguous ?

Reply:
Yes, any sliced array will be, 1D arrays will be both F and C contiugous (as well as arrays that can be "squeezed" into 1D or are empty, but both of those depend on `NPY_RELAXED_STRIDES_CHECKING`)

Reply:
Also: https://docs.scipy.org/doc/numpy/reference/arrays.ndarray.html#internal-memory-layout-of-an-ndarray although it only semi-covers all of this probably. May be best to just try around a bit looking at `arr.flags`, `arr[::2].flags`, `arr.T.flags`, etc.

Reply:
> Yes, any sliced array will be, 1D arrays will be both F and C contiugous (as well as arrays that can be "squeezed" into 1D or are empty, but both of those depend on NPY_RELAXED_STRIDES_CHECKING)

if an array is both F and C contiguous it will be still called "trivially iterable" right? I tried with sliced arrays and 1d arrays both are F and C contiguous.

I am wondering when the below will be false ?
```
    #define PyArray_TRIVIALLY_ITERABLE(arr) ( \
                        PyArray_NDIM(arr) <= 1 || \
                        PyArray_CHKFLAGS(arr, NPY_ARRAY_C_CONTIGUOUS) || \
                        PyArray_CHKFLAGS(arr, NPY_ARRAY_F_CONTIGUOUS) \
                        )
```

Reply:
Thanks for the links, I will try to look into it.

Reply:
You probably got it already, but: `np.ones(10)[::2].flags` is true, but `np.ones((10, 10))[::2, :].flags` is not.

Reply:
Thanks @seberg !